gateway/gateway.go

```diff
@@ -0,0 +1,144 @@
+package gateway
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"sync"
+
+	"github.com/Frontman-Labs/frontman/log"
+	"github.com/Frontman-Labs/frontman/config"
+	"github.com/Frontman-Labs/frontman/plugins"
+	"github.com/Frontman-Labs/frontman/service"
+)
+
+type APIGateway struct {
+	bs                 service.ServiceRegistry
+	plugs              []plugins.FrontmanPlugin
+	conf               *config.Config
+	clients            map[string]*http.Client
+	clientLock         sync.Mutex
+	currentTargetIndex int
+	log                log.Logger
+}
+
+func NewAPIGateway(bs service.ServiceRegistry, plugs []plugins.FrontmanPlugin, conf *config.Config, clients map[string]*http.Client, logger log.Logger) *APIGateway {
+	return &APIGateway{
+		bs:      bs,
+		plugs:   plugs,
+		conf:    conf,
+		clients: clients,
+		log: logger,
+	}
+}
+
+
+func (g *APIGateway) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	currentTargetIndex := 0
+	root := buildRoutes(g.bs.GetServices())
```

Comment:
did u accidently forget to add the `go refreshConnections`? anyways since its not a wrapper function anymore it should be placed somewhere else

Reply:
I did 😂